### PR TITLE
Misc fixes & cleanups

### PR DESCRIPTION
**Pull request by cboesch-gpsw**

### CODE REVIEW

File: libnodegl/hmap.c [hunk 1]
```diff
@@ -90,11 +90,27 @@ static struct hmap_ref ref_from_entry(const struct hmap *hm, const struct hmap_e
...
 }
 
-static struct hmap_entry *fixed_entry_from_ref(const struct hmap *hm, struct hmap_ref ref, struct hmap_ref removed)
+static struct hmap_entry *fixed_entry_from_ref(const struct hmap *hm, struct hmap_ref ref,
+                                               struct hmap_ref removed, int current_id)
 {
     if (!HAS_REF(ref))
         return NULL;
-    const int need_fix = ref.bucket_id == removed.bucket_id && ref.entry_id > removed.entry_id;
+    const int need_fix = \
```
Comment (mbouron-gpsw) [inline libnodegl/hmap.c:98]: Is this backslash really needed here ?

File: libnodegl/hmap.c [hunk 2]
```diff
@@ -90,11 +90,27 @@ static struct hmap_ref ref_from_entry(const struct hmap *hm, const struct hmap_e
...
+        ref.bucket_id == removed.bucket_id &&
+        /* Only the entries after the removed one are shifted */
+        ref.entry_id > removed.entry_id
+        /*
+         * The following last condition prevents from fixing the reference
+         * twice (and thus decrementing too much): if the passed reference
+         * `ref` (be it a next or prev, it doesn't matter) is located earlier
+         * in the current bucket, it means the complementary reference (prev if
+         * next, next if prev) in that particular entry was for sure already
+         * fixed.  Indeed, since the current entry points to it, it means that
```
Comment (mbouron-gpsw) [inline libnodegl/hmap.c:109]: Extra space.

File: libnodegl/hmap.c [hunk 3]
```diff
@@ -90,11 +90,27 @@ static struct hmap_ref ref_from_entry(const struct hmap *hm, const struct hmap_e
...
+         * the refered entry was also pointing to the current entry with its
```
Comment (mbouron-gpsw) [inline libnodegl/hmap.c:110]: referred

**Review by mbouron-gpsw — APPROVED** [review]
Other than my remarks, LGTM.

File: libnodegl/hmap.c (hunk 1 above)
Comment (cboesch-gpsw) [reply libnodegl/hmap.c]: Nope, removed.

File: libnodegl/hmap.c (hunk 2 above)
Comment (cboesch-gpsw) [reply libnodegl/hmap.c]: Removed.

File: libnodegl/hmap.c (hunk 3 above)
Comment (cboesch-gpsw) [reply libnodegl/hmap.c]: Fixed.

**Pull request merged**